src/Compilers/VisualBasic/Portable/Semantics/Conversions.vb

```diff
@@ -2053,7 +2077,16 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
 
             If sourceType Is Nothing Then
                 source = source.GetMostEnclosedParenthesizedExpression()
-                sourceType = If(source.Kind <> BoundKind.ArrayLiteral, source.Type, New ArrayLiteralTypeSymbol(DirectCast(source, BoundArrayLiteral)))
+                If source.Kind = BoundKind.ArrayLiteral Then
+                    sourceType = New ArrayLiteralTypeSymbol(DirectCast(source, BoundArrayLiteral))
```

Comment:
Looking at `ArrayLiteralTypeSymbol` there is some information about this in the comment. Makes me wonder why the same rationale isn't applicable to tuple literals.

---
In reply to: [76699033](https://github.com/dotnet/roslyn/pull/13335#discussion_r76699033) [](ancestors = 76699033)